op-node/rollup/sync/config.go

```diff
@@ -1,8 +1,55 @@
 package sync
 
+import "fmt"
+
+type Mode string
+
+// I'm currently using full sync & snap sync externally because that is what users tend to expect, but I
```

Comment:
I suspect this is going to cause a lot of confusion for users given that you will need to use `--syncmode=snap` on op-node with `--syncmode=full` on op-geth to get an archive node (particularly after 4844 starts being used).